src/utils/price.ts

```diff
@@ -0,0 +1,77 @@
+import BN from 'bn.js'
+import { assert } from './miscellaneous'
+import { TEN, UNLIMITED_ORDER_AMOUNT_BIGNUMBER } from 'const'
+import BigNumber from 'bignumber.js'
+
+export function adjustAmountToLowerPrecision(amount: BN, higherPrecision: number, lowerPrecision: number): BN {
+  if (higherPrecision === lowerPrecision) {
+    // no adjustment required
+    return amount
+  }
+
+  assert(higherPrecision > lowerPrecision, 'higherPrecision must be > lowerPrecision')
+
+  const difference = new BN(higherPrecision - lowerPrecision)
+
+  // divides amount by difference in precision, rounding. E.g:
```

Comment:
I love that you add these examples. Keep doing that. Code is easier to follow

Reply:
I'm selfish and do them for my self to understand what's going on :D